Source/WebCore/platform/graphics/texmap/TextureMapperLayer.cpp

```diff
@@ -80,6 +153,90 @@ TextureMapperLayer::~TextureMapperLayer()
     removeFromParent();
 }
 
+void TextureMapperLayer::processDescendantLayersFlatteningRequirements()
+{
+    // Traverse all children in depth first, post-order
+    for (auto* child : m_children) {
+        child->processDescendantLayersFlatteningRequirements();
+
+        if (child->flattensAsLeafOf3DSceneOr3DPerspective())
+            child->processFlatteningRequirements();
+    }
+}
+
+void TextureMapperLayer::processFlatteningRequirements()
+{
+    if (m_flattenedLayer)
+        m_flattenedLayer = nullptr;
+
+    // Reset transformations as this layer is root 2D
+    SetForScope scopedLocalTransform(m_layerTransforms.localTransform, TransformationMatrix());
+    m_layerTransforms.combined = { };
+    m_layerTransforms.combinedForChildren = { };
+#if USE(COORDINATED_GRAPHICS)
```

Comment:
Why is only this part protected by the guard?

Reply:
Windows port also uses TextureMapper but does not use COORDINATED_GRAPHICS.
future* transforms are specific to COORDINATED_GRAPHICS